packages/core/src/modules/proofs/services/ProofService.ts

```diff
@@ -558,6 +564,29 @@ export class ProofService {
     return proofRecord
   }
 
+  /**
+   * Process a received {@link PresentationProblemReportMessage}.
+   *
+   * @param messageContext The message context containing a presentation problem report message
+   * @returns proof record associated with the presentation acknowledgement message
+   *
+   */
+  public async processProblemReport(
+    messageContext: InboundMessageContext<PresentationProblemReportMessage>
+  ): Promise<ProofRecord> {
+    const { message: presentationProblemReportMessage, connection } = messageContext
+
+    this.logger.debug(`Processing presentation ack with id ${presentationProblemReportMessage.id}`)
+
+    const proofRecord = await this.getByThreadAndConnectionId(presentationProblemReportMessage.threadId, connection?.id)
+
+    // Update record
+    await this.updateState(proofRecord, ProofState.None)
+    proofRecord.errorMsg = `${presentationProblemReportMessage.description.code} : ${presentationProblemReportMessage.description.en}`
```

Comment:
If you add the errorMsg assignment to before the updateState call you don't need have to call `update` again. (the updateState will also store the record)